File: Cap xB/Calculadora V4/calculadoraV4.vb
Module CALCULADORA_V4
    Dim OPCAO As Char
    Dim R, A, B As Single
    Sub Main()
        OPCAO = "0"
        Do While (OPCAO <> "5")
            Console.WriteLine("CALCULADORA - V4")
            Console.WriteLine("================")
            Console.WriteLine("")
            Console.WriteLine("1 - Adicao")
            Console.WriteLine("2 - Subtracao")
            Console.WriteLine("3 - Multiplicacao")
            Console.WriteLine("4 - Divisao")
            Console.WriteLine("5 - Fim de Programa")
            Console.WriteLine("")           
            Console.Write("Escolha uma opcao: ")
            OPCAO = Console.ReadLine()
            If (OPCAO <> "5") Then
                Select Case OPCAO
                    Case "1"
                        ROTCALC("+")
                    Case "2"
                        ROTCALC("-")
                    Case "3"
                        ROTCALC("*")
                    Case "4"
                        ROTCALC("/")
                    Case Else
                        Console.WriteLine("")
                        Console.WriteLine("Opcao invalida - Tente novamente.")
                        Console.WriteLine("")
                End Select
            End If
        Loop
    End Sub

    Sub ENTRADA()
        Console.WriteLine("")
        Console.Write("Entre um valor para A: ")
        A = Console.ReadLine()
        Console.Write("Entre um valor para B: ")
        B = Console.ReadLine()
    End Sub

    Sub SAIDA()
        Console.WriteLine("")
        Console.Write("O resultado da operacao equivale a: ")
        Console.WriteLine(String.Format("{0:0.00}", R))
        Console.WriteLine("")
    End Sub

    Function CALCULO(ByVal X As Single, ByVal Y As Single, ByVal OPERADOR As Char) As Single
        Select Case OPERADOR
            Case "+"
                CALCULO = X + Y
            Case "-"
                CALCULO = X - Y
            Case "*"
                CALCULO = X * Y
            Case "/"
                CALCULO = X / Y
            Case Else
                Console.WriteLine("Operador inválido.")
                CALCULO = 0 
        End Select
    End Function

    Sub ROTCALC(ByVal OPERACAO As Char)
        Console.WriteLine("")
        Select Case (OPERACAO)
            Case "+"
                Console.WriteLine("Rotina de Adicao")
            Case "-"
                Console.WriteLine("Rotina de Subtracao")
            Case "*"
                Console.WriteLine("Rotina de Multiplicacao")
            Case "/"
                Console.WriteLine("Rotina de Divisao")
        End Select
        ENTRADA()
        If (OPERACAO = "/") Then
            If (B = 0) Then
                Console.WriteLine("")
                Console.WriteLine("O resultado da operacao equivale a: ERRO.")
                Console.WriteLine("")
            Else
                R = CALCULO(A, B, "/")
                SAIDA()
            End If
        End If
        If Not (OPERACAO = "/") Then
            R = CALCULO(A, B, OPERACAO)
            SAIDA()
        End If
    End Sub

End Module

File: Cap xB/Calculadora V4/calculadoraV4.lua
  function ENTRADA()
    print()
    io.write("Entre o 1o. valor: ") 
    A = io.read("*number")
    io.write("Entre o 2o. valor: ") 
    B = io.read("*number")
  end

  function SAIDA()
    print()
    print(string.format("O resultado da operacao equivale a:  %.2f.", R))
    print()
  end

  function CALCULO(X, Y, OPERADOR)
    if (OPERADOR == "+") then return X + Y end
    if (OPERADOR == "-") then return X - Y end
    if (OPERADOR == "*") then return X * Y end
    if (OPERADOR == "/") then return X / Y end
  end

  function ROTCALC(OPERACAO)
    print()
    if (OPERACAO == "+") then print("Rotina de Adicao")        end
    if (OPERACAO == "-") then print("Rotina de Subtracao")     end
    if (OPERACAO == "*") then print("Rotina de Multiplicacao") end
    if (OPERACAO == "/") then print("Rotina de Divisao")       end
    ENTRADA()
    if (OPERACAO == "/") then
      if (B == 0) then
        print()
        print("O resultado da operacao equivale a: ERRO.")
        print()
      else
        R = CALCULO(A, B, "/")
        SAIDA()
      end
    end
    if not (OPERACAO == "/") then
      R = CALCULO(A, B, OPERACAO)
      SAIDA()
    end
  end

while (OPCAO ~= 5) do
  print("CALCULADORA - V4")
  print("================")
  print()
  print("1 - Adicao")
  print("2 - Subtracao")
  print("3 - Multiplicacao")
  print("4 - Divisao")
  print("5 - Fim de Programa")
  print()
  io.write("Escolha uma opcao: ")
  OPCAO = io.read("*number")
  if (OPCAO ~= 5) then
    if (OPCAO == 1) then ROTCALC("+") end
    if (OPCAO == 2) then ROTCALC("-") end
    if (OPCAO == 3) then ROTCALC("*") end
    if (OPCAO == 4) then ROTCALC("/") end
    if (OPCAO < 1 or OPCAO > 5) then
      print()
      print("Opcao invalida - Tente novamente.")
      print()
    end
  end
end
